src/plugins/workspace/server/plugin.ts

```diff
@@ -115,12 +116,44 @@ export class WorkspacePlugin implements Plugin<WorkspacePluginSetup, WorkspacePl
       if (path === '/') {
         const workspaceListResponse = await this.client?.list(
           { request, logger: this.logger },
-          { page: 1, perPage: 1 }
+          { page: 1, perPage: 100 }
         );
+        const basePath = core.http.basePath.serverBasePath;
+
         if (workspaceListResponse?.success && workspaceListResponse.result.total > 0) {
-          return toolkit.next();
+          const workspaceList = workspaceListResponse.result.workspaces;
+          // If user only has one workspace, go to overview page of that workspace
+          if (workspaceList.length === 1) {
+            return response.redirected({
+              headers: {
+                location: `${basePath}/w/${workspaceList[0].id}/app/${WORKSPACE_NAVIGATION_APP_ID}`,
```

Comment:
Maybe we can try using the `formatUrlWithWorkspaceId` method.

Reply:
In the server side, the `formatUrlWithWorkspaceId ` method will be useless, because the parameter `basePath: IBasePath` is public side.

Reply:
I c, thanks for clarification.